ambry-frontend/src/main/java/com/github/ambry/frontend/AmbrySecurityService.java

```diff
@@ -135,6 +140,17 @@ public void postProcessRequest(RestRequest restRequest, Callback<Void> callback)
           exception = e;
         }
       }
+    } else if (restRequest.getRestMethod() == RestMethod.POST && storageQuotaService != null) {
```

Comment:
I think `AsyncRequestResponseHandler::handleRequest` might be a better place to handle the storage quota instead of AmbrySecurityService. The reason is primarily because this is comes earlier than `AmbrySecurityService::processRequest` in the request handling call chain. Also by the time code hits `handleRequest` the `RestRequest` object is already populated so account and container info should be already populated by then.
The reason I am nitpicking this is that if we have enough information to throttle request in  `AsyncRequestResponseHandler::handleRequest` (in form of `RestRequest` object passed to it), then there is no point of carrying request processing forward and adding it to the `AsyncRequestWorker` queue.

Reply:
This is ok since we need to get account and container id from the args, which is not available till PostBlobHandler.